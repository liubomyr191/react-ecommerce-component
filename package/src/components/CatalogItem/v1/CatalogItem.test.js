import React from "react";
import renderer from "react-test-renderer";
import mockComponents from "../../../tests/mockComponents";
import CatalogItem from "./CatalogItem";

test("CatalogItem basic snapshot", () => {
  const component = renderer.create((
    <CatalogItem
      components={mockComponents}
      placeholderImageURL="/resources/placeholder.gif"
      product={{
        title: "Basic Reaction Product",
        slug: "basic-reaction-product",
        vendor: "Example Manufacturer",
        primaryImage: {
          URLs: {
            large: "/images/responsive-sticker/large.jpg",
            medium: "/images/responsive-sticker/medium.jpg",
            small: "/images/responsive-sticker/small.png",
            thumbnail: "/images/responsive-sticker/thumbnail.png"
          }
        },
        pricing: [{
          currency: {
            code: "USD"
          },
          displayPrice: "$12.99 - $19.99"
        }],
        isSoldOut: false,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: false,
        isBestseller: true
      }}
      currencyCode="USD"
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("CatalogItem with placeholder image", () => {
  const component = renderer.create((
    <CatalogItem
      components={mockComponents}
      placeholderImageURL="/resources/placeholder.gif"
      product={{
        title: "Basic Reaction Product",
        slug: "basic-reaction-product",
        vendor: "Example Manufacturer",
        primaryImage: null,
        pricing: [{
          currency: {
            code: "USD"
          },
          displayPrice: "$12.99 - $19.99"
        }],
        isSoldOut: false,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: false,
        isBestseller: true
      }}
      currencyCode="USD"
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
