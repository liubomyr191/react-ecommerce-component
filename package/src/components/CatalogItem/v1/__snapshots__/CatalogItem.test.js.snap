// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogItem basic snapshot 1`] = `
<div>
  Link(undefined)
</div>
`;

exports[`CatalogItem with placeholder image 1`] = `
<div>
  Link(undefined)
</div>
`;
